Packs/CommonScripts/Scripts/script-ShowOnMap.yml

```diff
@@ -4,43 +4,65 @@ commonfields:
 name: ShowOnMap
 script: >
   var lat = parseFloat(args.lat);
-
   var lng = parseFloat(args.lng);
+  var address = args.address;
+
+  if (isNaN(lat) || isNaN(lng)) {
+    if (!address || address.trim().length === 0){
+    return {
+        ContentsFormat: formats.text,
+        Type: entryTypes.error,
+        Contents: "Please enter either coordinates (lat,lng) or an address (which requires a GoogleMaps integration instance)."
+    };
 
+    }
+    // An active instance of GoogleMaps exists, geocoding is possible
+      var commandResult = executeCommand("google-maps-geocode", {"search_address" : address});
+      var ec = commandResult[0].EntryContext["GoogleMaps(val.lat && val.lat == obj.lat && val.lng && val.lng == obj.lng)"];
 
-  if(isNaN(lat) || isNaN(lng)) {
-      return {
-          ContentsFormat: formats.text,
-          Type: entryTypes.error,
-          Contents: 'Invalid coordinates'
-      };
+      if (ec)
+      {
+          lat = ec.lat;
+          lng = ec.lng;
+      }
+      else { // EntryContext could not be parsed correctly
+          return {
+              ContentsFormat: formats.text,
+              Type: entryTypes.error,
+              Contents: "Geocoding failed."
+          };
+      }
   }
 
+  //  lat,lng are not null (whether they were valid arguments or geocoded)
 
   var locationObj = {
-      lat: lat,
-      lng: lng
+    lat: lat,
+    lng: lng
   };
 
-
   return {
-      // type
-      Type: entryTypes.map,
-       // contents
-      ContentsFormat: formats.json, Contents: locationObj
+    Type: entryTypes.map,
+    ContentsFormat: formats.json,
+    Contents: locationObj
   };
+
 type: javascript
 tags:
 - Utility
-comment: Returns a map entry with marker on the given coordinate (lat, lng).
+comment: Returns a map entry with marker on the given coordinates (lat, lng), or address (requires a configured GoogleMaps instance)
 enabled: true
 args:
 - name: lat
-  required: true
+  required: false
   default: true
   description: Latitude (e.g. 32.064622)
 - name: lng
-  required: true
+  required: false
   description: Longitude (e.g. 34.774131)
+- name: address
+  default: true
+  required: false
+  description: Address (will be converted to coordinates and then shown)
```

Comment:
No need to mention the logic of the address, but do give an example address, such as `Mountain View, California`.
Please make sure to match the description in the readme.